src/MPQEditor/MPQCircleSelector.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from "vscode";
+
+import {
+  CircleGraphCtrl,
+  CircleGraphEvent,
+} from "../CircleGraph/CircleGraphCtrl";
+
+export interface MPQSelectionEvent {
+  onSelection(names: string[], document: vscode.TextDocument): void;
+  onClosed(panel: vscode.WebviewPanel): void;
+  onOpened(panel: vscode.WebviewPanel): void;
+}
+
+export type MPQSelectionCmdOpenArgs = {
+  modelPath: string;
+  document: vscode.TextDocument;
+  names: any;
+  panel: vscode.WebviewPanel;
+};
+
+export class MPQSelectionPanel
+  extends CircleGraphCtrl
+  implements CircleGraphEvent
+{
+  public static readonly viewType = "one.viewer.mpq";
```

Comment:
I wonder if it's 'viewer' or' 'editor'..?

Reply:
@dayo09 
For now its viewer with 'selector' abilities. I mean it doesn't have a document to edit. IMHO.

Reply:
Oh, Thanks for the explanation!